sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveTableScanExec.scala

```diff
@@ -233,7 +234,10 @@ case class GpuHiveTableScanExec(requestedAttributes: Seq[Attribute],
     val requestedCols = requestedAttributes.filter(a => !partitionKeys.contains(a.name))
                                            .toList
     val distinctColumns = requestedCols.distinct
-    val distinctFields  = distinctColumns.map(a => tableSchema.apply(a.name))
+    // In hive column names are case-insensitive but the default tableSchema lookup is
+    // case-sensitive
+    val fieldMap = CaseInsensitiveMap(tableSchema.map(f => (f.name, f)).toMap)
```

Comment:
What happens when `spark.sql.caseSensitive` is set to true?

Reply:
Turns out that Hive is always case-insensitive So even if I try to create a table with two columns with different case I get an error from hive.

```
scala> spark.conf.set("spark.sql.caseSensitive", true)

scala> spark.sql("""create table testcase_text(id int, nAme string, Name string)""").collect
24/08/15 15:38:30 WARN ResolveSessionCatalog: A Hive serde table will be created as there is no table provider specified. You can set spark.sql.legacy.createHiveTableByDefault to false so that native data source table will be created instead.
org.apache.spark.sql.AnalysisException: org.apache.hadoop.hive.ql.metadata.HiveException: org.apache.hadoop.hive.ql.metadata.HiveException: Duplicate column name name in the table definition.
  at org.apache.spark.sql.hive.HiveExternalCatalog.withClient(HiveExternalCatalog.scala:110)
  at org.apache.spark.sql.hive.HiveExternalCatalog.createTable(HiveExternalCatalog.scala:244)
  at org.apache.spark.sql.catalyst.catalog.ExternalCatalogWithListener.createTable(ExternalCatalogWithListener.scala:94)
```

If I try to put in a name with the wrong case in the query when case sensitive is true, then spark outputs an error in the logical plan phase before the GPU code ever runs.